### PR TITLE
feat: Jupyterhub cognito fine grained access control

### PR DESCRIPTION
**Pull request by dorukozturk**
### What does this PR do?

This PR creates a cognito identity pool, creates an iam role that is tied to the cognito user and injects credentials to the spawner using auth hooks. This way customers can manage all of their users through cognito and assign iam roles to the notebooks themselves for principle of least privilege. 

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: ai-ml/jupyterhub/cognito.tf [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+          "s3:prefix": "$${aws:PrincipalTag/department}"
+        }
+      }
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_cognito_identity_pool_provider_principal_tag" "example" {
```
Comment (askulkarni2) [inline ai-ml/jupyterhub/cognito.tf:194]: Create only if cognito is used.

File: ai-ml/jupyterhub/helm/jupyterhub/jupyterhub-values-cognito.yaml [hunk 2]
```diff
@@ -23,7 +24,43 @@ hub:
   extraConfig:
     jupyterhub_config.py: |-
       c.KubeSpawner.start_timeout = 1200
+      c.Authenticator.enable_auth_state = True
 
+    cognito_config.py: |-
```
Comment (askulkarni2) [inline ai-ml/jupyterhub/helm/jupyterhub/jupyterhub-values-cognito.yaml:29]: Very cool!!

@lusoal this is similar to what we discussed at KubeCon.

**Review by askulkarni2 — COMMENTED** [review]
@dorukozturk PR is fantastic and much needed by customers. 

Just some minor notes but otherwise looking great.

**Comment by dorukozturk** [comment]
I added the conditionals ! Thanks for all the kind words and fast reviews :)

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Comment by dorukozturk** [comment]
Fixed the index errors

**Pull request merged**